### PR TITLE
ECER-2192: Update list and view of message to include thread information

### PR DESCRIPTION
**Pull request by qsl-deepak**
---
name: ECER-2192: Update list and view of message to include thread information
about: Update list and view of message to include thread information
---

## Title
ECER-2192: Update list and view of message to include thread information

## Description

- Merge "Master" to "Stories/ECER-2192" branch
- Added "do not reply" check to message thread
- Re-factored "Reply to Message" component to call backend API to get data thus making the component "refresh safe"
- Re-factored "MessageList" component to use Vue js options API 
- Re-factored - "Mark message as read" functionality

## Related Jira Issue(s)

- ECER-{###}
- etc.

## Checklist
- [x] I have tested these changes locally.
- [x] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

## Screenshots (if applicable)

image

## Additional Comments (optional)
Any additional context or information that might be helpful for the reviewer.

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/message.ts [hunk 1]
```diff
@@ -37,20 +27,9 @@ export const useMessageStore = defineStore("message", {
     messageById(id: string): Components.Schemas.Communication | undefined {
       return this.messages?.find((message) => message.id === id);
     },
-    async fetchMessages() {
-      this.messages = (await getMessages()).data;
-    },
+
     async markMessageAsRead(messageId: string) {
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/message.ts:31]: We can probably just call this API method directly and remove from store?

**Review by pcatkins — APPROVED** [review]
FE looks good, just one small _optional_ comment

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/message.ts (hunk 1 above)
Comment (qsl-deepak) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/message.ts]: Yup, Done..

**Review by farzadnadiri — DISMISSED** [review]

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/MessageList.vue [hunk 2]
```diff
@@ -1,43 +1,72 @@
...
+      const params = {
+        page,
+        pageSize: PAGE_SIZE,
+      };
+      const response = await getMessages(params);
+      this.messages = response.data?.communications || [];
+      this.messageCount = response.data?.totalMessagesCount || 0;
+    },
+    async resetMessageStore() {
+      this.messageStore.currentMessage = null;
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/MessageList.vue:67]: If you are reseting the whole store you can just use `this.messageStore.$reset()` and skip this method

**Review by pcatkins — APPROVED** [review]

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/MessageList.vue (hunk 2 above)
Comment (qsl-deepak) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/MessageList.vue]: Done..

**Review by pcatkins — APPROVED** [review]

**Pull request merged**